xmtp_mls/src/storage/encrypted_store/group.rs

```diff
@@ -126,40 +129,56 @@ impl DbConnection {
         created_before_ns: Option<i64>,
         limit: Option<i64>,
         conversation_type: Option<ConversationType>,
+        consent_state: Option<ConsentState>,
     ) -> Result<Vec<StoredGroup>, StorageError> {
-        let mut query = dsl::groups.order(dsl::created_at_ns.asc()).into_boxed();
+        use crate::storage::schema::consent_records::dsl as consent_dsl;
+        use crate::storage::schema::groups::dsl as groups_dsl;
+
+        let mut query = groups_dsl::groups
+            .order(groups_dsl::created_at_ns.asc())
+            .into_boxed();
 
         if let Some(allowed_states) = allowed_states {
-            query = query.filter(dsl::membership_state.eq_any(allowed_states));
+            query = query.filter(groups_dsl::membership_state.eq_any(allowed_states));
         }
 
         if let Some(created_after_ns) = created_after_ns {
-            query = query.filter(dsl::created_at_ns.gt(created_after_ns));
+            query = query.filter(groups_dsl::created_at_ns.gt(created_after_ns));
         }
 
         if let Some(created_before_ns) = created_before_ns {
-            query = query.filter(dsl::created_at_ns.lt(created_before_ns));
-        }
-
-        if let Some(limit) = limit {
-            query = query.limit(limit);
+            query = query.filter(groups_dsl::created_at_ns.lt(created_before_ns));
         }
 
         if let Some(conversation_type) = conversation_type {
             match conversation_type {
                 ConversationType::Group => {
-                    query = query.filter(dsl::dm_inbox_id.is_null());
+                    query = query.filter(groups_dsl::dm_inbox_id.is_null());
                 }
                 ConversationType::Dm => {
-                    query = query.filter(dsl::dm_inbox_id.is_not_null());
+                    query = query.filter(groups_dsl::dm_inbox_id.is_not_null());
                 }
                 ConversationType::Sync => {}
             }
         }
 
-        query = query.filter(dsl::purpose.eq(Purpose::Conversation));
+        if let Some(state) = consent_state {
+            query =
+                query
+                    .inner_join(consent_dsl::consent_records.on(
+                        sql::<diesel::sql_types::Text>("hex(groups.id)").eq(consent_dsl::entity),
+                    ))
+                    .filter(consent_dsl::state.eq(state))
+                    .select(groups_dsl::groups::all_columns());
+        }
```

Comment:
@insipx pulled this into a separate PR so it wasn't blocking the message order stuff. Just FYI.